tests/unit/test_serialization.py

```diff
@@ -267,105 +375,167 @@ def assert_almost_equal(value: np.ndarray, reference: Any):
         np.testing.assert_almost_equal(value, reference)
 
 
-def assert_scenario_correct(
-    deserialized_dataset: Mapping[ComponentType, np.ndarray],
+def assert_single_dataset_entries(
+    deserialized_dataset: SingleDataset,
     serialized_dataset: Mapping[str, Any],
     sparse_components: list[ComponentType],
+    data_filter,
 ):
+    """Check a SingleDataset's individual components for correctness"""
+    # TODO What is the purpose of this check? Implement without using derserialized_dataset
     for key in serialized_dataset["data"]:
-        if key not in deserialized_dataset:
+        if key not in deserialized_dataset and not is_filtered_out(data_filter, key):
             assert key in sparse_components
+            assert False
 
-    for component, component_result in deserialized_dataset.items():
-        assert isinstance(component_result, np.ndarray)
-
-        component_input = serialized_dataset["data"][component]
-        assert len(component_result) == len(component_input)
-        for result_entry, input_entry in zip(component_result, component_input):
-            if isinstance(input_entry, dict):
-                for attr, values in input_entry.items():
-                    assert attr in result_entry.dtype.names
-                    assert_almost_equal(result_entry[attr], values)
+    for component, component_input in serialized_dataset["data"].items():
+        if is_filtered_out(data_filter, component):
+            assert component not in deserialized_dataset
+            continue
+
+        component_result = deserialized_dataset[component]
+        assert isinstance(component_input, list)  # TODO Why?
+
+        # Complete component data checks
+        if is_columnar_filter(data_filter, component):
+            assert isinstance(component_result, dict)
+            assert all(len(v) == len(component_input) for v in component_result.values())
+        else:
+            assert isinstance(component_result, np.ndarray)
+            assert len(component_result) == len(component_input)
+
+        # Individual data entry checks
+        for comp_idx, input_entry in enumerate(component_input):
+            if is_non_compact_list(input_entry):
+                for attr in input_entry:
+                    if is_columnar_filter(data_filter, component):
+                        if is_filtered_out(data_filter, component, attr):
+                            assert attr not in component_result
+                            continue
+                        assert attr in component_result.keys()
+                        assert_almost_equal(component_result[attr][comp_idx], component_input[comp_idx][attr])
```

Comment:
Nitpick: Maybe change names of `component_result` and `component_input` to reflect that the second comes from deserialized data and the first from serialized one?